File: core/src/main/java/org/runetranscriber/core/cirth/example/CirthEreborExample2.java
package org.runetranscriber.core.cirth.example;

import org.runetranscriber.core.AbstractExample;
import org.runetranscriber.core.FontLetterList;
import org.runetranscriber.core.LanguageLetterList;
import org.runetranscriber.core.Phoneme;
import org.runetranscriber.core.PhonemeList;
import org.runetranscriber.core.RuneList;
import org.runetranscriber.core.cirth.CerthRune;
import org.runetranscriber.core.cirth.CerthFontLetter;

/**
 * <a href="http://ring-lord.tripod.com/cirth/ex_moria2.htm">Balin's tomb, lower inscription</a>
 */
public final class CirthEreborExample2 extends AbstractExample<CerthRune, CerthFontLetter>
{
    @Override
    protected FontLetterList<CerthFontLetter> createFontLetters()
    {
        final CerthFontLetter[] myLetters = {
                // balin
                CerthFontLetter.C2, CerthFontLetter.C48, CerthFontLetter.C31, CerthFontLetter.C39,
                CerthFontLetter.C53,
                CerthFontLetter.SPACE,
                // son
                CerthFontLetter.C34, CerthFontLetter.C56, CerthFontLetter.C53,
                CerthFontLetter.SPACE,
                // of
                CerthFontLetter.C50, CerthFontLetter.C4,
                CerthFontLetter.SPACE,
                // fundin
                CerthFontLetter.C3, CerthFontLetter.C42, CerthFontLetter.C33, CerthFontLetter.C39, CerthFontLetter.C53,
                CerthFontLetter.SPACE,
                // lord
                CerthFontLetter.C31, CerthFontLetter.C50, CerthFontLetter.C12, CerthFontLetter.C9,
                CerthFontLetter.SPACE,
                // of
                CerthFontLetter.C50, CerthFontLetter.C4, CerthFontLetter.SPACE,
                // moria
                CerthFontLetter.C6, CerthFontLetter.C50, CerthFontLetter.C12, CerthFontLetter.C39, CerthFontLetter.C48, };

        return new FontLetterList<CerthFontLetter>(myLetters);
    }

    @Override
    protected LanguageLetterList createLanguageLetters()
    {
        final String text = "balin son of fundin lord of moria";

        return new LanguageLetterList(text);
    }

    @Override
    protected PhonemeList createPhonemes()
    {
        final Phoneme[] myPhonemes = {
                // balin
                Phoneme.B, Phoneme.A, Phoneme.L, Phoneme.I, Phoneme.N, Phoneme.SPACE,
                // son
                Phoneme.S, Phoneme.SCHWA2, Phoneme.N, Phoneme.SPACE,
                // of
                Phoneme.O, Phoneme.V, Phoneme.SPACE,
                // fundin
                Phoneme.F, Phoneme.U, Phoneme.ND, Phoneme.I, Phoneme.N, Phoneme.SPACE,
                // lord
                Phoneme.L, Phoneme.O, Phoneme.R, Phoneme.D, Phoneme.SPACE,
                // of
                Phoneme.O, Phoneme.V, Phoneme.SPACE,
                // moria
                Phoneme.M, Phoneme.O, Phoneme.R, Phoneme.I, Phoneme.A };

        return new PhonemeList(myPhonemes);
    }

    @Override
    protected RuneList<CerthRune> createRunes()
    {
        final CerthRune[] myRunes = {
                // balin
                CerthRune.C2, CerthRune.C48, CerthRune.C31, CerthRune.C39, CerthRune.C53, CerthRune.SPACE,
                // son
                CerthRune.C34, CerthRune.C56, CerthRune.C53, CerthRune.SPACE,
                // of
                CerthRune.C50, CerthRune.C4, CerthRune.SPACE,
                // fundin
                CerthRune.C3, CerthRune.C42, CerthRune.C33, CerthRune.C39, CerthRune.C53, CerthRune.SPACE,
                // lord
                CerthRune.C31, CerthRune.C50, CerthRune.C12, CerthRune.C9, CerthRune.SPACE,
                // of
                CerthRune.C50, CerthRune.C4, CerthRune.SPACE,
                // moria
                CerthRune.C6, CerthRune.C50, CerthRune.C12, CerthRune.C39, CerthRune.C48, };

        return new RuneList<CerthRune>(myRunes);
    }
}

File: webapp/src/main/js/example/CirthEreborExample2.js
/*
 * <a href="http://ring-lord.tripod.com/cirth/ex_moria2.htm">Balin's tomb, lower inscription</a>
 */
define(
        [ "CerthRune" ],
        function(CerthRune)
        {
            "use strict";
            var CirthEreborExample2 =
            {
                NAME: "Cirth Erebor Example 2",

                IMAGE: "https://rawgit.com/jmthompson2015/runetranscriber/master/core/doc/images/lordoftherings/TLOTR_BalinsTomb_CirthErebor.jpg",

                LANGUAGE_STRING: "balin son of fundin lord of moria",

                LANGUAGE_WORDS: [ "balin", " ", "son", " ", "of", " ", "fundin", " ", "lord", " ", "of", " ", "moria" ],

                PHONEMES: [ CerthRune.properties["c2"].phoneme, CerthRune.properties["c48"].phoneme,
                        CerthRune.properties["c31"].phoneme, CerthRune.properties["c39"].phoneme,
                        CerthRune.properties["c53"].phoneme, CerthRune.properties["space"].phoneme,
                        CerthRune.properties["c34"].phoneme, CerthRune.properties["c56"].phoneme,
                        CerthRune.properties["c53"].phoneme, CerthRune.properties["space"].phoneme,
                        CerthRune.properties["c50"].phoneme, CerthRune.properties["c4"].phoneme,
                        CerthRune.properties["space"].phoneme, CerthRune.properties["c3"].phoneme,
                        CerthRune.properties["c42"].phoneme, CerthRune.properties["c33"].phoneme,
                        CerthRune.properties["c39"].phoneme, CerthRune.properties["c53"].phoneme,
                        CerthRune.properties["space"].phoneme, CerthRune.properties["c31"].phoneme,
                        CerthRune.properties["c50"].phoneme, CerthRune.properties["c12"].phoneme,
                        CerthRune.properties["c9"].phoneme, CerthRune.properties["space"].phoneme,
                        CerthRune.properties["c50"].phoneme, CerthRune.properties["c4"].phoneme,
                        CerthRune.properties["space"].phoneme, CerthRune.properties["c6"].phoneme,
                        CerthRune.properties["c50"].phoneme, CerthRune.properties["c12"].phoneme,
                        CerthRune.properties["c39"].phoneme, CerthRune.properties["c48"].phoneme ],

                RUNES: [
                // balin
                CerthRune.C2, CerthRune.C48, CerthRune.C31, CerthRune.C39, CerthRune.C22, CerthRune.SPACE,
                // son
                CerthRune.C35, CerthRune.C56, CerthRune.C22, CerthRune.SPACE,
                // of
                CerthRune.C50, CerthRune.C4, CerthRune.SPACE,
                // fundin
                CerthRune.C3, CerthRune.C42, CerthRune.C33, CerthRune.C39, CerthRune.C22, CerthRune.SPACE,
                // lord
                CerthRune.C31, CerthRune.C50, CerthRune.C12, CerthRune.C9, CerthRune.SPACE,
                // of
                CerthRune.C50, CerthRune.C4, CerthRune.SPACE,
                // moria
                CerthRune.C6, CerthRune.C50, CerthRune.C12, CerthRune.C39, CerthRune.C48, ],

                FONT_LETTERS: [
                // balin
                CerthRune.properties["c2"].fontLetter, CerthRune.properties["c48"].fontLetter,
                        CerthRune.properties["c31"].fontLetter, CerthRune.properties["c39"].fontLetter,
                        CerthRune.properties["c22"].fontLetter, CerthRune.properties["space"].fontLetter,
                        // son
                        CerthRune.properties["c35"].fontLetter, CerthRune.properties["c56"].fontLetter,
                        CerthRune.properties["c22"].fontLetter, CerthRune.properties["space"].fontLetter,
                        // of
                        CerthRune.properties["c50"].fontLetter, CerthRune.properties["c4"].fontLetter,
                        CerthRune.properties["space"].fontLetter,
                        // fundin
                        CerthRune.properties["c3"].fontLetter, CerthRune.properties["c42"].fontLetter,
                        CerthRune.properties["c33"].fontLetter, CerthRune.properties["c39"].fontLetter,
                        CerthRune.properties["c22"].fontLetter, CerthRune.properties["space"].fontLetter,
                        // lord
                        CerthRune.properties["c31"].fontLetter, CerthRune.properties["c50"].fontLetter,
                        CerthRune.properties["c12"].fontLetter, CerthRune.properties["c9"].fontLetter,
                        CerthRune.properties["space"].fontLetter,
                        // of
                        CerthRune.properties["c50"].fontLetter, CerthRune.properties["c4"].fontLetter,
                        CerthRune.properties["space"].fontLetter,
                        // moria
                        CerthRune.properties["c6"].fontLetter, CerthRune.properties["c50"].fontLetter,
                        CerthRune.properties["c12"].fontLetter, CerthRune.properties["c39"].fontLetter,
                        CerthRune.properties["c48"].fontLetter ],
            };

            if (Object.freeze)
            {
                Object.freeze(CirthEreborExample2);
            }

            return CirthEreborExample2;
        });
